lib/ReactViews/Map/Panels/SharePanel/BuildShareLink.js

```diff
@@ -16,12 +16,13 @@ const userPropWhiteList = ['hideExplorerPanel', 'activeTabId'];
  * Builds a share link that reflects the state of the passed Terria instance.
  *
  * @param terria The terria instance to serialize.
+ * @param viewState the viewState to read whether we're viewing the catalog or not
  * @returns {String} A URI that will rebuild the current state when viewed in a browser.
  */
-export function buildShareLink(terria) {
+export function buildShareLink(terria, viewState) {
```

Comment:
Not super convinced that we need to introduce `viewState` into BuildShareLink, though i guess non of these will be problem once we are in mobx land

Reply:
We need some way to tell if the explorer window is open via `explorerPanelIsVisible` so I can't think of any other way since there's no reference from `viewState` -> `terria`? So yes hopefully mobx land simplifies this